cmake/HalideTestHelpers.cmake

```diff
@@ -26,12 +26,14 @@ if (NOT TARGET Halide::Test)
 
     # Tests are built with the equivalent of OPTIMIZE_FOR_BUILD_TIME (-O0 or /Od).
     # Also allow tests, via conditional compilation, to use the entire
-    # capability of the CPU being compiled on via -march=native. This
-    # presumes tests are run on the same machine they are compiled on.
+    # capability of the CPU being compiled on via -march=native if the
+    # the compiler supports it. This also presumes tests are run on the
+    # same machine they are compiled on.
     target_compile_options(Halide_test INTERFACE
                            $<$<CXX_COMPILER_ID:MSVC>:/Od>
                            $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-O0>
-                           $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-march=native>)
```

Comment:
This seems to be turning off -march=native for clang everywhere. Did clang remove the flag or am I misunderstanding the change?

The decision on whether to use this flag has a bit of history and it could go either way, but it seems we should just remove it entirely if we're going to turn it off for clang. (The issue is it chooses to use the architecture of the host compilation system, which may be a higher rev of the instruction set than the one the tests run on. This makes the tests track the hardware they're run on, which is sometimes an advantage in better coverage and sometimes an inconvenience in debugging a test failure between systems.)

Reply:
It doesn't affect the JIT runtime, though. So having a `host` target is different from `-march=native`. One could add clang back to the list via `$<$<CXX_COMPILER_ID:Clang>:-march=native>`, noting that Clang and AppleClang are _not_ the same compiler as far as CMake is concerned.